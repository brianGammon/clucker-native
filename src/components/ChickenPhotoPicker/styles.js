import { StyleSheet } from 'react-native';
import variables from 'native-base/dist/src/theme/variables/platform';

export default StyleSheet.create({
  container: {
    marginTop: 20,
    padding: 10,
  },
  photoControls: {
    flex: 1,
  },
  label: {
    fontSize: 17,
    paddingBottom: 10,
    color: variables.inputColorPlaceholder,
  },
});
