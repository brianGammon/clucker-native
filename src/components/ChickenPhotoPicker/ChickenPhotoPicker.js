/* @flow */
import React from 'react';
import {
  View, Button, Text, Image,
} from 'react-native';
import styles from './styles';

type Props = {
  photoUrl: string,
  newImage: {
    mime: string,
    data: string,
  } | null,
  originalPhotoUrl: string,
  onRemoveProfilePhoto: () => void,
  onResetProfilePhoto: () => void,
  onSelectPhoto: (withCamera: boolean) => void,
};

const ChickenPhotoPicker = ({
  photoUrl,
  newImage,
  originalPhotoUrl,
  onRemoveProfilePhoto,
  onResetProfilePhoto,
  onSelectPhoto,
}: Props) => {
  let imageSource = require('../../assets/default-profile-photo.png');
  if (photoUrl !== '') {
    imageSource = { uri: photoUrl };
  }
  if (newImage) {
    imageSource = {
      uri: `data:${newImage.mime};base64,${newImage.data}`,
    };
  }
  return (
    <View style={styles.container}>
      <Text style={styles.label}>Profile Photo:</Text>
      <View style={{ flexDirection: 'row' }}>
        <Image style={{ width: 200, height: 200 }} source={imageSource} />
        <View style={styles.photoControls}>
          {photoUrl !== ''
            && !newImage && (
              <Button onPress={onRemoveProfilePhoto} title="Remove" />
          )}
          {((originalPhotoUrl !== '' && photoUrl === '') || newImage) && (
            <Button onPress={onResetProfilePhoto} title="Reset" />
          )}
          <Button onPress={() => onSelectPhoto(false)} title="Select Photo" />
          <Button onPress={() => onSelectPhoto(true)} title="Take Photo" />
        </View>
      </View>
    </View>
  );
};

export default ChickenPhotoPicker;
