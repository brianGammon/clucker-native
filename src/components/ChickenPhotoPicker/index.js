import ChickenPhotoPicker from './ChickenPhotoPicker';

export default ChickenPhotoPicker;
