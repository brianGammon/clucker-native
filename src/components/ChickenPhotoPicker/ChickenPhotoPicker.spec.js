import React from 'react';
import ChickenPhotoPicker from '.';

const props = {
  photoUrl: '',
  newImage: null,
  originalPhotoUrl: '',
  onRemoveProfilePhoto: () => {},
  onResetProfilePhoto: () => {},
  onSelectPhoto: () => {},
};

test('ChickenPhotoPicker renders when no previous photo and no new photo selected', () => {
  const wrapper = shallow(<ChickenPhotoPicker {...props} />);
  expect(wrapper).toMatchSnapshot();
});

test('ChickenPhotoPicker renders when existing photo was removed', () => {
  const wrapper = shallow(
    <ChickenPhotoPicker {...{ ...props, originalPhotoUrl: '/test' }} />,
  );
  expect(wrapper).toMatchSnapshot();
});

test('ChickenPhotoPicker renders when original photo exists, new photo selected', () => {
  const wrapper = shallow(
    <ChickenPhotoPicker
      {...{
        ...props,
        photoUrl: '/test',
        newImage: { mime: 'jpeg', data: 'testdata' },
      }}
    />,
  );
  expect(wrapper).toMatchSnapshot();
});

test('ChickenPhotoPicker renders when originalPhotoUrl exists, and new image not selected', () => {
  const wrapper = shallow(
    <ChickenPhotoPicker
      {...{
        ...props,
        photoUrl: '/test',
        originalPhotoUrl: '/test',
      }}
    />,
  );
  expect(wrapper).toMatchSnapshot();
});
