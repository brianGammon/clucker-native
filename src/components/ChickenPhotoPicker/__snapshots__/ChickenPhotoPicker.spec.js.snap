// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChickenPhotoPicker renders when existing photo was removed 1`] = `
<Component
  style={
    Object {
      "marginTop": 20,
      "padding": 10,
    }
  }
>
  <Component
    style={
      Object {
        "color": "#575757",
        "fontSize": 17,
        "paddingBottom": 10,
      }
    }
  >
    Profile Photo:
  </Component>
  <Component
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/default-profile-photo.png",
        }
      }
      style={
        Object {
          "height": 200,
          "width": 200,
        }
      }
    />
    <Component
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Button
        onPress={[Function]}
        title="Reset"
      />
      <Button
        onPress={[Function]}
        title="Select Photo"
      />
      <Button
        onPress={[Function]}
        title="Take Photo"
      />
    </Component>
  </Component>
</Component>
`;

exports[`ChickenPhotoPicker renders when no previous photo and no new photo selected 1`] = `
<Component
  style={
    Object {
      "marginTop": 20,
      "padding": 10,
    }
  }
>
  <Component
    style={
      Object {
        "color": "#575757",
        "fontSize": 17,
        "paddingBottom": 10,
      }
    }
  >
    Profile Photo:
  </Component>
  <Component
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/default-profile-photo.png",
        }
      }
      style={
        Object {
          "height": 200,
          "width": 200,
        }
      }
    />
    <Component
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Button
        onPress={[Function]}
        title="Select Photo"
      />
      <Button
        onPress={[Function]}
        title="Take Photo"
      />
    </Component>
  </Component>
</Component>
`;

exports[`ChickenPhotoPicker renders when original photo exists, new photo selected 1`] = `
<Component
  style={
    Object {
      "marginTop": 20,
      "padding": 10,
    }
  }
>
  <Component
    style={
      Object {
        "color": "#575757",
        "fontSize": 17,
        "paddingBottom": 10,
      }
    }
  >
    Profile Photo:
  </Component>
  <Component
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "data:jpeg;base64,testdata",
        }
      }
      style={
        Object {
          "height": 200,
          "width": 200,
        }
      }
    />
    <Component
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Button
        onPress={[Function]}
        title="Reset"
      />
      <Button
        onPress={[Function]}
        title="Select Photo"
      />
      <Button
        onPress={[Function]}
        title="Take Photo"
      />
    </Component>
  </Component>
</Component>
`;

exports[`ChickenPhotoPicker renders when originalPhotoUrl exists, and new image not selected 1`] = `
<Component
  style={
    Object {
      "marginTop": 20,
      "padding": 10,
    }
  }
>
  <Component
    style={
      Object {
        "color": "#575757",
        "fontSize": 17,
        "paddingBottom": 10,
      }
    }
  >
    Profile Photo:
  </Component>
  <Component
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "/test",
        }
      }
      style={
        Object {
          "height": 200,
          "width": 200,
        }
      }
    />
    <Component
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Button
        onPress={[Function]}
        title="Remove"
      />
      <Button
        onPress={[Function]}
        title="Select Photo"
      />
      <Button
        onPress={[Function]}
        title="Take Photo"
      />
    </Component>
  </Component>
</Component>
`;
